File: leetcode/string/leetcode_1328.java
package leetcode.string;

public class leetcode_1328 {
    public static void main(String[] args) {
        leetcode_1328 lc = new leetcode_1328();

        System.out.println(lc.breakPalindrome("aa"));
    }

    public String breakPalindrome(String palindrome) {
        char[] palindromeArr = palindrome.toCharArray();

        for (int i = 0; i < palindromeArr.length / 2; i++) {
            if (palindromeArr[i] != 'a') {
                palindromeArr[i] = 'a';
                return String.valueOf(palindromeArr);
            }
        }

        palindromeArr[palindromeArr.length - 1] = 'b';
        return palindrome.length() < 2 ? "" : String.valueOf(palindromeArr);
    }
}

File: string/LeetCode_1328.kt
package string

class LeetCode_1328 {

    fun breakPalindrome(palindrome: String): String {
        if (palindrome.length < 2) {
            return ""
        }

        repeat (palindrome.length / 2) {
            if (palindrome[it] > 'a') {
                return palindrome.substring(0, it) + 'a' + palindrome.substring(it + 1, palindrome.length)
            }
        }

        return palindrome.substring(0, palindrome.lastIndex) + 'b'
    }
}

fun main() {
    // Input
    val palindrome = "abccba"
    // Output
    LeetCode_1328().breakPalindrome(palindrome).let {
        println(it)
        require(it == "aaccba")
    }
}
